core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -442,9 +445,13 @@ object SparkEnv extends Logging {
     val addedJarsAndFiles = (addedJars ++ addedFiles).map((_, "Added By User"))
     val classPaths = (addedJarsAndFiles ++ classPathEntries).sorted
 
+    // Add Hadoop properties, it will ignore configs including in Spark
+    val hadoopProperties = hadoopConf.asScala.filter(entry => !conf.contains("spark.hadoop." + entry.getKey)).
```

Comment:
instead of conf.contains, use startsWith?

Reply:
I actually don't think we should filter them - if I am trying to debug why the config is not getting picked up it's useful to see everyone listed here

Reply:
@felixcheung Yes, I argee with you. I will remove it.